app/components/UI/FiatOnRampAggregator/components/TransactionDetails.tsx

```diff
@@ -113,27 +107,69 @@ const renderStage = (stage: string, paymentType: string) => {
 				</>
 			);
 		}
+		default: {
+			return (
+				<>
+					<Image source={failedIcon} />
+					<Text bold big primary centered style={styles.stageDescription}>
+						{stage === 'failed' ? strings('fiat_on_ramp_aggregator.transaction.failed') : 'cancelled'}
+					</Text>
+					<Text small centered style={styles.stageMessage}>
+						{strings('fiat_on_ramp_aggregator.transaction.failed_description')}
+					</Text>
+				</>
+			);
+		}
 	}
 };
 
 interface Props {
+	/**
+	 * Object that represents the current route info like params passed to it
+	 */
 	order: any;
+	/**
+	 * Current Network provider
+	 */
+	provider: any;
+	/**
+	 * Frequent RPC list from PreferencesController
+	 */
+	frequentRpcList: any;
 }
 
-const TransactionDetails: React.FC<Props> = ({ order }: Props) => {
+const TransactionDetails: React.FC<Props> = ({ order, provider, frequentRpcList }: Props) => {
+	const { data, state, createdAt, amount, cryptoFee, cryptoAmount, currencySymbol, currency, txHash } = order;
 	const { colors } = useTheme();
+	const explorer = useBlockExplorer(provider, frequentRpcList);
 	const styles = createStyles(colors);
-	const date = toDateFormat(order.data.createdAt);
-
+	const date = toDateFormat(createdAt);
+	const amountOut = Number(amount) - Number(cryptoFee);
+	const exchangeRate = Number(amountOut) / Number(cryptoAmount);
+	const handleLinkPress = useCallback(async (url: string) => {
+		const supported = await Linking.canOpenURL(url);
+		if (supported) {
+			await Linking.openURL(url);
+		}
+	}, []);
 	return (
 		<View>
-			<View style={styles.stage}>{renderStage(order.state, order.data.paymentMethod.name)}</View>
+			<View style={styles.stage}>
+				<Stage stage={state} paymentType={data.paymentMethod?.name} />
+			</View>
 			<Text centered primary style={styles.tokenAmount}>
-				{order.data.cryptoAmount}
+				{renderFromTokenMinimalUnit(
+					toTokenMinimalUnit(
+						cryptoAmount,
+						order.cryptoCurrency?.decimals || data.cryptoCurrency?.decimals
+					).toString(),
+					data.cryptoCurrency?.decimals
+				)}{' '}
+				{data.cryptoCurrency?.symbol}
 			</Text>
 			<Text centered small style={styles.fiatColor}>
-				{order.data.fiatCurrency.denomSymbol}
-				{order.data.fiatAmount} {order.data.fiatCurrency.symbol}
+				{currencySymbol}
+				{renderFiat(amountOut, currency, order.fiatCurrency?.decimals || data.fiatCurrency?.decimals)}
```

Comment:
`order` does not have `fiatCurrency` property. We should rely on `order.data.fiatCurrency.decimals`